### PR TITLE
III-6271 InvalidArgumentException on Empty Category

### PR DESCRIPTION
**Pull request by JonasVHG**
### Added

- `EmptyCategoryId`: Create new Exception for when `CategoryId` is empty.
- `event-with-empty-term-id.json`: new data file to be used in acceptance tests

### Changed

- `CategoryID`: throw new Exception `EmptyCategoryId`, when Category is Empty.
- `ImportTermRequestBodyParser`: catch new `EmptyCategoryId`
- `ImportTermRequestBodyParserTest`:  Add Unit Test with empty `Category`.
- `terms.feature`:  Add acceptance test with empty `Category`.

### Fixed

- Cleaner error when an empty Category is posted.

---

Ticket: https://jira.publiq.be/browse/III-6271

### CODE REVIEW

**Review by grubolsch — APPROVED** [review]

File: src/Http/Import/ImportTermRequestBodyParser.php [hunk 1]
```diff
@@ -58,7 +59,11 @@ function ($term) use ($ignoreDomains) {
                 function ($term, int $index) {
                     if ($term instanceof stdClass && isset($term->id) && is_string($term->id)) {
                         $id = $term->id;
-                        $category = $this->categoryResolver->byId(new CategoryID($id));
+                        try {
+                            $category = $this->categoryResolver->byId(new CategoryID($id));
+                        } catch (EmptyCategoryId $exception) {
+                            $category = null;
```
Comment (LucWollants) [inline src/Http/Import/ImportTermRequestBodyParser.php:65]: Should we catch this? We once discussed that we would not silently accept wrong values or modify them to correct values. We should always inform the API integrator with a problem report.

Comment (JonasVHG) [reply src/Http/Import/ImportTermRequestBodyParser.php:65]: They will get a 4xx Error Response, instead of a 5xx Server Error.
`The term  does not exist or is not supported` Same as one uses  a termid that does not exist(e.g., a term with id 123)

File: tests/Http/Import/ImportTermRequestBodyParserTest.php [hunk 2]
```diff
@@ -214,4 +214,37 @@ public function it_throws_on_unsupported_term(): void
...
+                    'domain' => 'eventtype',
+                ],
+            ],
+        ];
+
+        $request = (new Psr7RequestBuilder())
+            ->withParsedBody($terms)
+            ->build('POST');
+
+        $this->assertCallableThrowsApiProblem(
```
Comment (LucWollants) [inline tests/Http/Import/ImportTermRequestBodyParserTest.php:240]: This is not what we see inside the Sentry issue

File: src/Http/Import/ImportTermRequestBodyParser.php (hunk 1 above)
Comment (LucWollants) [reply src/Http/Import/ImportTermRequestBodyParser.php]: As discussed we now report a more specific error about an empty category ID

**Review by LucWollants — APPROVED** [review]
